### DISCUSSION TITLE
IndexError: list index out of range

### DISCUSSION THREAD

**Discussion by marius-mathies** [tuonglab/dandelion#395] [General]
Hello, I want to run dandelion on my 10X data (5') and already in the beginning I get the error message "IndexError: list index out of range"

Does anyone know how to deal with that?

Best,
Marius

(base) [mam8432@compute-a-16-83 Normal_01]$ singularity run -B /home/mam8432/Normal_01 /n/app/singularity/containers/mam8432/sc-dandelion-latest.sif dandelion-preprocess
Software versions:

dandelion==0.3.7 pandas==2.2.2 numpy==1.26.4 matplotlib==3.8.4 networkx==3.3 scipy==1.13.0

Begin preprocessing

command line parameters:
: --------------------------------------------------------------
    --meta = None
    --chain = ig
    --org = human
    --file_prefix = all
    --db = imgt
    --strain = None
    --sep = _
    --flavour = strict
    --filter_to_high_confidence = False
    --keep_trailing_hyphen_number = False
    --skip_format_header = False
    --skip_tigger = False
    --skip_reassign_dj = False
    --skip_correct_c = False
    --clean_output = False
: --------------------------------------------------------------

Formatting fasta(s) : |          | 0/0 [00:00<?, ?it/s]                                                                                                                                        
Assigning genes : |          | 0/0 [00:00<?, ?it/s]                                                                                                                                            
Processing data file(s) : |          | 0/0 [00:00<?, ?it/s]                                                                                                                                    
Traceback (most recent call last):
  File "/share/dandelion_preprocess.py", line 378, in <module>
    main()
  File "/share/dandelion_preprocess.py", line 309, in main
    ddl.pp.reassign_alleles(
  File "/opt/conda/envs/sc-dandelion-container/lib/python3.11/site-packages/dandelion/preprocessing/_preprocessing.py", line 1508, in reassign_alleles
    Path(filepathlist_heavy[0]),
         ~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

**Comment by zktuong** [comment]
hi @MariusMathies, it looks like it has failed at the very first step (formating fasta). can you check if there's any incorrect file names or weird hidden files in your folder?

**Reply by marius-mathies** [reply]
I sued the wrong prefix. Thank you for the help!

**Reply by liezeltamon** [reply]
Sorry @zktuong just saw this discussion, this is related to what I posted here, #117 Singularity error - But my prefix is correct, also checked for hidden files...

**Reply by liezeltamon** [reply]
Okay it worked for me when I used a CSV file for that 1 sample!
